deployments/traefik_forward_auth.go

```diff
@@ -0,0 +1,485 @@
+					"app.kubernetes.io/instance": "traefik-forward-auth",
+					"app.kubernetes.io/name":     "traefik-forward-auth",
+					"app.kubernetes.io/version":  "2.2.0",
+				},
+			},
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Name:     "http",
+						Port:     4181,
+						Protocol: "TCP",
+						TargetPort: intstr.IntOrString{
+							Type:   intstr.String,
+							StrVal: "http"},
+					},
+				},
+				Selector: map[string]string{
+					"app.kubernetes.io/instance": "traefik-forward-auth",
+					"app.kubernetes.io/name":     "traefik-forward-auth",
+				},
+			},
+		},
+		metav1.CreateOptions{})
+
+	return err
+}
+
+func (k TraefikForwardAuth) createOauthClientSecret(ctx context.Context, c *kubernetes.Cluster) error {
+	passwordAuth, err := auth.RandomPasswordAuth()
+	if err != nil {
+		return err
+	}
+
+	_, err = c.Kubectl.CoreV1().Secrets(TraefikForwardAuthDeploymentID).Create(ctx,
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-client", TraefikForwardAuthDeploymentID),
+			},
+			StringData: map[string]string{
+				"username": passwordAuth.Username,
+				"password": passwordAuth.Password,
+			},
+			Type: "Opaque",
+		}, metav1.CreateOptions{})
+
+	return err
+}
+
+func (k TraefikForwardAuth) createDeployment(ctx context.Context, c *kubernetes.Cluster, options kubernetes.InstallationOptions) error {
+	replicas := int32(1)
+	domain, err := options.GetString("system_domain", "")
+	if err != nil {
+		return err
+	}
+
+	cookieSecret, err := randstr.Hex16()
+	if err != nil {
+		return err
+	}
+
+	_, err = c.Kubectl.AppsV1().Deployments(TraefikForwardAuthDeploymentID).Create(ctx, &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: TraefikForwardAuthDeploymentID,
+			Labels: map[string]string{
+				"app.kubernetes.io/instance": TraefikForwardAuthDeploymentID,
+				"app.kubernetes.io/name":     TraefikForwardAuthDeploymentID,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/instance": TraefikForwardAuthDeploymentID,
+					"app.kubernetes.io/name":     TraefikForwardAuthDeploymentID,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: TraefikForwardAuthDeploymentID,
+					Labels: map[string]string{
+						"app.kubernetes.io/instance": TraefikForwardAuthDeploymentID,
+						"app.kubernetes.io/name":     TraefikForwardAuthDeploymentID,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "epinio-ca-root",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "epinio-ca-root",
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "traefik-forward-auth",
+							Image: "docker.io/thomseddon/traefik-forward-auth:" + TraefikForwardAuthVersion,
```

Comment:
Aha! ;)
I'm missing some documentation/note pointing me to https://github.com/thomseddon/traefik-forward-auth
Maybe in a later PR

Reply:
I added a bullet in the Acceptance Criteria in #667